### DISCUSSION TITLE
How to make git repo info retrieval like stars conditional to cookie consent?

### DISCUSSION THREAD

**Discussion by fwendland** [squidfunk/mkdocs-material#4329] [Q&A] [answered]
Hi, thanks for the great work.

I'm using the built-in git repository feature to include a nice link to my project's GitHub repo. To some extend, I like the additional info -- latest tag, stars and forks -- displayed with the link to my repo. However, I have the problem that it's an external request. I would like to make this request conditional to a cookie consent.

Is it possible to make the info retrieval for git repos conditional to some cookie consent? How would I go about implementing it?

**Comment by squidfunk** [comment, marked as answer]
You're right, it should be under consent, so the fact that it's not is actually an oversight. You can create a new issue. We should conditionally request URLs only if the user gave his consent. We should tie this to a [new cookie](https://squidfunk.github.io/mkdocs-material/setup/ensuring-data-privacy/#consent-cookies) (all of those are essentially no cookies but local storage entries) named `github`, that is automatically added when a repository was configured, and which the user can disable in the settings of the consent form.
